code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -282,340 +285,264 @@ GLOBAL_VAR_INIT(dynamic_forced_storyteller, null)
-
-// Yes, this is copy pasted from game_mode
-/datum/game_mode/dynamic/check_finished(force_ending)
-	if(!SSticker.setup_done || !gamemode_ready)
-		return FALSE
-	if(replacementmode && round_converted == 2)
-		return replacementmode.check_finished()
-	if(SSshuttle.emergency && (SSshuttle.emergency.mode == SHUTTLE_ENDGAME))
-		return TRUE
-	if(station_was_nuked)
-		return TRUE
-	if(force_ending)
-		return TRUE
-	for(var/datum/dynamic_ruleset/rule in executed_rules)
-		if(rule.flags & HIGHLANDER_RULESET)
-			return rule.check_finished()
-
-/datum/game_mode/dynamic/proc/log_threat(var/log_str,var/verbose = FALSE)
-	threat_log_verbose += ("[worldtime2text()]: "+log_str)
-	SSblackbox.record_feedback("tally","dynamic_threat_log",1,log_str)
-	if(!verbose)
-		threat_log += log_str
+	priority_announce("A summary has been copied and printed to all communications consoles.", "Enemy communication intercepted. Security level elevated.", "intercept")
+	if(GLOB.security_level < SEC_LEVEL_BLUE)
+		set_security_level(SEC_LEVEL_BLUE)
 
 /datum/game_mode/dynamic/proc/show_threatlog(mob/admin)
 	if(!SSticker.HasRoundStarted())
-		alert("The round hasn't started yet!")
+		tgui_alert(usr, "The round hasn't started yet!")
 		return
 
 	if(!check_rights(R_ADMIN))
 		return
 
-	var/list/out = list("<TITLE>Threat Log</TITLE><B><font size='3'>Threat Log</font></B><br><B>Starting Threat:</B> [initial_threat_level]<BR>")
+	var/list/out = list("<TITLE>Threat Log</TITLE><B><font size='3'>Threat Log</font></B><br><B>Starting Threat:</B> [threat_level]<BR>")
 
-	for(var/entry in threat_log_verbose)
+	for(var/entry in threat_log)
 		if(istext(entry))
 			out += "[entry]<BR>"
 
-	out += "<B>Remaining threat/threat_level:</B> [threat]/[threat_level]"
+	out += "<B>Remaining threat/threat_level:</B> [mid_round_budget]/[threat_level]"
 
 	usr << browse(out.Join(), "window=threatlog;size=700x500")
 
 /// Generates the threat level using lorentz distribution and assigns peaceful_percentage.
 /datum/game_mode/dynamic/proc/generate_threat()
-	var/relative_threat = LORENTZ_DISTRIBUTION(GLOB.dynamic_curve_centre, GLOB.dynamic_curve_width)
-	threat_level = round(lorentz_to_threat(relative_threat), 0.1)
-
-	peaceful_percentage = round(LORENTZ_CUMULATIVE_DISTRIBUTION(relative_threat, GLOB.dynamic_curve_centre, GLOB.dynamic_curve_width), 0.01)*100
-
-	SSblackbox.record_feedback("tally","dynamic_threat",threat_level,"Initial threat level")
-	SSblackbox.record_feedback("tally","dynamic_threat",GLOB.dynamic_curve_centre,"Curve centre")
-	SSblackbox.record_feedback("tally","dynamic_threat",GLOB.dynamic_curve_width,"Curve width")
-	SSblackbox.record_feedback("tally","dynamic_threat",peaceful_percentage,"Percent of same-vote rounds that are more peaceful")
```

Comment:
add these back, after peaceful_percentage is assigned, though change "same-vote" to "same-center"

Reply:
Added and tweaked